d2l-icon-button-shared-styles.html

```diff
@@ -28,26 +29,36 @@
 				};
 			}
 
-			:host(::-moz-focus-inner) {
+			.d2l-icon-button::-moz-focus-inner {
 				border: 0;
 			}
-			:host {
+			.d2l-icon-button {
 				@apply(--d2l-icon-button);
 			}
-			:host(:disabled){
+			.d2l-icon-button:disabled {
 				opacity: 0.5;
 			}
-			:host(:hover),
-			:host(:focus) {
+			.d2l-icon-button:hover,
+			.d2l-icon-button:focus {
 				@apply(--d2l-icon-button-hover);
 			}
-			:host d2l-icon {
+			.d2l-icon-button > d2l-icon {
 				@apply(--d2l-icon-button-icon);
 			}
-			:host(:hover) d2l-icon,
-			:host(:focus) d2l-icon {
+			.d2l-icon-button:hover > d2l-icon,
+			.d2l-icon-button:focus > d2l-icon {
 				@apply(--d2l-icon-button-icon-hover);
 			}
+
+			.d2l-icon-button > span {
+				@apply(--d2l-offscreen);
+			}
+			:host-context([dir="rtl"]) .d2l-icon-button > span {
+				@apply(--d2l-offscreen-rtl);
+			}
```

Comment:
In some browsers (Firefox I think), if one selector is invalid, the whole block is ignored.